core/src/main/scala/tech/beshu/ror/accesscontrol/utils/ClaimsOps.scala

```diff
@@ -54,52 +77,83 @@ class ClaimsOps(val claims: Claims) extends Logging {
       .fold(_ => NotFound, identity)
   }
 
-  def groupsClaim(claimName: Jwt.ClaimName): ClaimSearchResult[UniqueList[Group]] = {
-    claimName.name.read[Any](claims)
-      .map {
-        case value: String =>
-          Found(UniqueList.fromIterable((value :: Nil).flatMap(toGroup)))
-        case collection: java.util.Collection[_] =>
-          Found {
-            UniqueList.fromIterable {
-              collection.asScala
-                .collect {
-                  case value: String => value
-                  case value: Long => value.toString
-                }
-                .flatMap(toGroup)
-            }
+  def groupsClaim(groupIdsClaimName: Jwt.ClaimName,
+                  groupNamesClaimName: Option[Jwt.ClaimName]): ClaimSearchResult[UniqueList[Group]] = {
+
+    (for {
+      groupIds <- readGroupIds(groupIdsClaimName)
+      groupNames <- readGroupNames(groupNamesClaimName)
+      searchResult = groupIds match {
+        case Found(ids) => ClaimSearchResult.Found {
+          groupNames match {
+            case Found(names) if names.size == ids.size =>
```

Comment:
I wonder if we shouldn't treat the case where `names.size != ids.size` as sth that we should warn a user? For sure this is not what is expected.

Reply:
I added debug logs describing what's happening.